schemas/common_definitions.json

```diff
@@ -349,24 +350,100 @@
           "additionalProperties": false
         },
         "repeat": {
-          "type": "object",
-          "properties": {
-            "answer_id": {
-              "type": "string"
-            },
-            "goto": {
-              "type": "string"
-            },
-            "type": {
-              "enum": [
-                "answer_value",
-                "answer_count",
-                "answer_count_minus_one"
-              ]
-            }
-          },
-          "required": [
-            "answer_id"
+          "oneOf": [
+              {
+                "type": "object",
+                "properties": {
+                  "type": {
+                    "enum": [
+                      "until"
+                    ]
+                  },
+                  "when": {
```

Comment:
Can we not use the existing `when` definition to reduce duplication?
`"$ref": "common_definitions.json#/when"`

Reply:
I'd have liked to, but the `when` in repeats only supports a subset of the usual behaviour, only one condition is allowed and only based on an `id`.

Reply:
my concern is that we will have 2 definitions of `when` and they could then deviate from each other and it would get confusing.

Reply:
what do you suggest instead?

Reply:
I think it would be better to reuse the existing definition but then limit the usage of it in `validator.py`